### ISSUE TITLE
YAML frontmatter parsed totally wrong when published to github pages via gh-deploy

### ISSUE THREAD

**Issue by CouldBeThis** [CouldBeThis/duckduckLib#23]
Must add YAML frontmatter to .md files in order to help Typora to put images in the right place. 

see: [Copy image files to target folder when inserting local image](https://support.typora.io/Images/#copy-image-files-to-target-folder-when-inserting-local-image)

So put frontmatter like this in all documents:

```yaml
title: Instructions
description: How to install, use and uninstalled DuckDuckLib Firefox extension.
authors:
	- CouldBeThis
date: 2021-03-15
# use for root-level document:
typora-copy-images-to: /images
# use for 2ndary level document:
#typora-copy-images-to: ../images
```

However it interprets all the lines as titles, or something, and it's a big mess. 

![image](https://user-images.githubusercontent.com/53547181/111811831-db9ab100-88ad-11eb-9fc9-fc9914ffb523.png)

Only when built with `mkdocs gh-deploy`, when served locally all is well.

**Comment by CouldBeThis** [comment]
Looking through [extensions](https://github.com/Python-Markdown/markdown/wiki/Third-Party-Extensions) and [plugins](https://github.com/mkdocs/mkdocs/wiki/MkDocs-Plugins) for something that seems like it might address it, found:

- [python-markdown-full-yaml-metadata](https://github.com/sivakov512/python-markdown-full-yaml-metadata) was able to install and include without build errors but not sure what do do beyond that. Doesn't automagically solve anything.

**Comment by CouldBeThis** [comment]
Looking through [extensions](https://github.com/Python-Markdown/markdown/wiki/Third-Party-Extensions) and [plugins](https://github.com/mkdocs/mkdocs/wiki/MkDocs-Plugins) for something that seems like it might address it, found:

`mkdcomments`

github: [ryneeverett/python-markdown-comments](https://github.com/ryneeverett/python-markdown-comments)

> Removes triple-dashed html comments (`<!---comment-->`) in preprocessing.

So after some experimentation, this seems to remove the junk from the outputted webpage:

```yaml
---
title: Instructions
description: How to install, use and uninstalled DuckDuckLib Firefox extension.
authors:
	- CouldBeThis
date: 2021-03-15
<!----
# use for root-level document:
typora-copy-images-to: /images
# use for 2ndary level document:
#typora-copy-images-to: ../images
---->
---
```

Though there is the question of whether the rest of the frontmatter is still working as intended. Wasn't replying on it for anything so will consider the matter dealt with for the time being.

**Comment by CouldBeThis** [comment]
fixed in 21ff41c
